components/check-buttons/view.tsx

```diff
@@ -0,0 +1,31 @@
+import { MouseEvent } from "react";
+
+import Button from "components/button";
+
+import { SelectOption } from "types/utils";
+
+interface CheckButtonsViewProps {
+  options: SelectOption[];
+  getButtonColor: (value: SelectOption) => string;
+  onClick: (value: SelectOption) => (e: MouseEvent<HTMLButtonElement>) => void;
+}
+
+export default function CheckButtonsView({
+  options,
+  getButtonColor,
+  onClick,
+}: CheckButtonsViewProps) {
+  return(
+    <div className="d-flex">
+      {options.map((opt, index) => 
+        <Button
+          key={`${opt.label}-${index}`}
+          color={getButtonColor(opt)}
```

Comment:
we can simply make two props (selectedColor, color) with default values to `primary` and that gray and then just check for option-selection for selected or not (which is logic that belongs to view-land)